shared/lib_weatherfile.cpp

```diff
@@ -1365,6 +1377,11 @@ bool weatherfile::open(const std::string &file, bool header_only)
 					continue;
 				}
 
+                if (m_columns[HOUR].data[i] > 23)
+                {
+                    subtract_hour = true;
+                }
+
 				if (m_columns[MINUTE].data[i] > 59)
```

Comment:
Sorry I'm late to this discussion. During implementation of the time-steppable PV model, I added a function to make sure hours were between 0-23. https://github.com/NREL/ssc/commit/01ef286ea05211ab21390d64cfe2ba999481eb36#diff-2ad11e7c78af69e327ea269d6cfea2c8 Is this partially duplicating the behavior?

Reply:
It looks like there are two paths for solar resource data, named "solar_resource_file" and "solar_resource_data" in the vartable. "solar_resource_data" was making the adjustment properly prior to this PR, but "solar_resource_file" was not. My debugger never went through common.cpp, so I ignored that code when doing this work, hence the duplicate function.

I'm open to having one function for both, but would prefer it ended up in lib_weatherfile.

Reply:
There's the same function for both weather data and weatherfiles: https://github.com/NREL/ssc/blob/32f4bcb69a05681866b7a4358bfb921895c1ad63/shared/lib_weatherfile.cpp#L438 implemented differently since the two classes organize their data differently.

Perhaps moving `start_hours_at_0` to be called earlier like you are doing here?